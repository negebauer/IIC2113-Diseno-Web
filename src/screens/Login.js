import React, { Component } from 'react'
import { connect } from 'react-redux'

import { devlog } from '../utils/log'
import { login, saveUser } from '../redux/modules/user'

const mapStateToProps = () => ({})

const mapDispatchToProps = {
  login,
  saveUser,
}

class Home extends Component {
  constructor(props) {
    super(props)
    this.state = {
      mail: '',
      password: '',
    }
  }
  submit = e => {
    e.preventDefault()
    devlog(this.state.mail)
    devlog(this.state.password)
    this.props.saveUser(this.state)
    this.props.login(this.state)
  }

  handleChange = ev =>
    this.setState({
      [ev.target.name]: Number(ev.target.value) || ev.target.value,
    })

  render() {
    devlog('Login', this.props)
    return (
      <div id="center30">
        <div id="login" className="row">
          <form className="col s12" ref={f => (this.form = f)}>
            <h2 className="header">Login</h2>
            <div className="row">
              <div className="input-field col s12">
                <input
                  id="mail"
                  type="email"
                  className="validate"
                  name="mail"
                  value={this.state.mail}
                  onChange={this.handleChange}
                />
                <label htmlFor="email">Email</label>
              </div>
            </div>
            <div className="row">
              <div className="input-field col s12">
                <input
                  id="password"
                  type="password"
                  className="validate"
                  name="password"
                  value={this.state.password}
                  onChange={this.handleChange}
                />
                <label htmlFor="password">Password</label>
              </div>
            </div>
            <button
              className="waves-effect waves-light btn-large"
              onClick={this.submit}
            >
              Login
            </button>
          </form>
        </div>
      </div>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
