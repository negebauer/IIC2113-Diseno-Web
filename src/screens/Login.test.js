import React from 'react'
import createHistory from 'history/createBrowserHistory'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Login from './Login'
import configureStore from '../redux/store'

let history
let initialState
let store

beforeEach(() => {
  history = createHistory()
  initialState = {}
  const api = { login: () => {}, saveUser: () => {}, replace: () => {} }
  store = configureStore(initialState, { history, api })
})

it('shallow renders without crashing', () => {
  shallow(<Login store={store} />)
})

it('mount renders without crashing', () => {
  mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Login />
      </ConnectedRouter>
    </Provider>
  )
})

it('matches snapshot', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Login />
      </ConnectedRouter>
    </Provider>
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders children', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Login />
      </ConnectedRouter>
    </Provider>
  )
  expect(wrapper.children().exists()).toBeTruthy()
})
