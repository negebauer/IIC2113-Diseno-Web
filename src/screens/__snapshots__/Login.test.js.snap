// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectedRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Login)>
        <Login
          api_key=""
          error=""
          loading={false}
          login={[Function]}
          replace={[Function]}
          saveUser={[Function]}
        >
          <div
            className="container"
          >
            <div
              className="row"
              id="login"
            >
              <form
                className="col s12"
              >
                <h2
                  className="header"
                >
                  Login
                </h2>
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <input
                      className="validate"
                      id="mail"
                      name="mail"
                      onChange={[Function]}
                      type="email"
                      value=""
                    />
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="input-field col s12"
                  >
                    <input
                      className="validate"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                  </div>
                </div>
                <button
                  className="waves-effect waves-light btn-large"
                  onClick={[Function]}
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </Login>
      </Connect(Login)>
    </Router>
  </ConnectedRouter>
</Provider>
`;
